### DISCUSSION TITLE
Edge weights in GIN

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2155] [Q&A] [answered]
How can I use edge weights with GINConv? I saw that GCN conv has an optional attribute which allows you to use graph weights. Is it not possible to do the same with GINConv?

**Comment by reshinthadithyan** [comment, marked as answer]
I guess, you can use [torch_geometric.nn.conv.GINEConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GINEConv).

**Reply by mdanb** [reply]
Thanks didn't notice that! haha

**Reply by simon-forb** [reply]
Note that the suggested model would not use the edge weights as weights but rather as attributes. What this means is that the edge attributes are added to the node attributes in the message passing step. Naturally one would expect that the edge weights would serve as a scaling factor, e.g., if an edge has weight $1/2$, then the node attributes from the connected node would by scaled by $1/2$. 

For instance see the implementation in DGL: https://docs.dgl.ai/en/0.8.x/generated/dgl.nn.pytorch.conv.GINConv.html

**Reply by rusty1s** [reply]
It does not make much sense to add `edge_weight` support to `GINConv` since this conflicts with the formulation of GIN in the first place (injectivity is no longer preserved).
